sphinx/themes/basic/static/doctools.js

```diff
@@ -264,6 +264,8 @@ var Documentation = {
   hideSearchWords : function() {
     $('#searchbox .highlight-link').fadeOut(300);
     $('span.highlighted').removeClass('highlighted');
+    newurl = window.location.href.replace(new RegExp('\\?highlight=[^#]*'), '');
+    history.pushState({}, null, newurl);
```

Comment:
This creates a new history entry, which makes sense, but moving back and forth in the history does *not* restore the highlighting. In #9551, I'm suggesting to use `history.replaceState()` instead.

Reply:
All right, I do support your change in #9551 and I've just removed the history change in this pull request.